static/app/components/compactSelect/control.tsx

```diff
@@ -59,7 +60,16 @@ export const SelectContext = createContext<SelectContextValue>({
   overlayIsOpen: false,
 });
 
-export interface ControlProps extends UseOverlayProps {
+export interface ControlProps
+  extends Omit<
+      React.BaseHTMLAttributes<HTMLDivElement>,
+      // omit keys from SingleListProps because those will be passed to <List /> instead
+      keyof Omit<
+        SingleListProps<React.Key>,
+        'children' | 'items' | 'grid' | 'compositeIndex' | 'label'
+      >
+    >,
+    UseOverlayProps {
```

Comment:
Adding `React.BaseHTMLAttributes<HTMLDivElement>` so we can attach HTML props, like `onKeyDown`, to `CompactSelect`. These props will be passed to the `ControlWrap` element.